themes/default/content/blog/cloud-systems-part-two/index.md

```diff
@@ -0,0 +1,220 @@
+
+## Prerequisites:
+
+- [Pulumi account](https://app.pulumi.com)
+
+- [Docker](https://www.docker.com/products/docker-desktop)
+
+- Python3
+
+Before we go into containerizing and deploying our website, let’s make it a little bit more useful. For this, we’re going to be using Flask, a lightweight web framework for Python. Fork and clone [this GitHub repository](https://github.com/katcosgrove/cloud-systems-101) to get the full sample code for part two of this series.
+
+## Expanding the Website
+
+Our file structure gets more complicated now that we are building a more dynamic website with server-side routing and throwing it into a Docker container. You will have something like this:
+
+```
+container-tutorial
+├── Pulumi.yaml
+├── requirements.txt
+├── __main__.py
+└── website
+    ├── Dockerfile
+    ├── requirements.txt
+    └── app
+        ├── server.py
+        ├── static
+        │   ├── normalize.css
+        │   ├── style.css
+        │   └── background.jpg
+        └── templates
+            ├── base.html
+            ├── index.html
+            ├── portfolio.html
+            └── about.html
+```
+
+Everything contained within the `app` directory is the website itself. We now have server-side routing, and a few different pages. To run it locally and see what the new website looks like, `pip3 install flask` to install the Flask web framework and then run `python3 server.py` from the `app` directory. Flask will return an IP address where your website is running; navigate there in your browser and click around a bit!
+
+`server.py`
+
+```python
+from flask import Flask, render_template
+
+app = Flask(__name__)
+
+@app.route("/")
+def home():
+    return render_template("index.html")
+
+@app.route("/about")
+def about():
+    return render_template("about.html")
+
+@app.route('/portfolio')
+def portfolio():
+    return render_template("portfolio.html")
+
+
+if __name__ == '__main__':
+    app.run(host='0.0.0.0', port=80, debug=True)
+```
+
+Where before our entire website was made up of a single HTML file, we now have several, and we're using Flask to handle the routing between them. In this server file, three routs are defined: home, about, and portfolio. Flask handles serving these up for us, and associating a route with a particular HTML file. This way, we get a nice `www.mywebsite.com/about` URL instead of something like `www.mywebsite.com/about.html`. Building a website this way also means we have the ability to apply some logic to each of these routes, such as adding database interaction, user login, and passing conditional variables from the server to the templates that will be rendering each page. We aren't doing any of that yet, but we will!
```

Comment:
```suggestion
Where before our entire website was made up of a single HTML file, we now have several, and we're using Flask to handle the routing between them. In this server file, three routes are defined: `home`, `about`, and `portfolio`. Flask handles serving these up for us and associating a route with a particular HTML file. This way, we get a nice `www.mywebsite.com/about` URL instead of something like `www.mywebsite.com/about.html`. Building a website this way also means we have the ability to apply some logic to each of these routes, such as adding database interaction, user login, and passing conditional variables from the server to the templates that will be rendering each page. We aren't doing any of that yet, but we will!
```